src/scripts/main.js

```diff
@@ -1 +1,61 @@
 'use strict';
+
+const success = (result) => {
+  const message = document.createElement('div');
+
+  message.className = 'success';
+  message.dataset.qa = 'notification';
+  message.innerText = result;
+  document.body.append(message);
+};
+
+const error = (result) => {
+  const message = document.createElement('div');
+
+  message.className = 'warning';
+  message.dataset.qa = 'notification';
+  message.innerText = result;
+  document.body.append(message);
+};
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', () => {
+    resolve('First promise was resolved');
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+  });
+
+  document.addEventListener('contextmenu', () => {
+    resolve('Second promise was resolved');
+  });
+});
+
+const thirdPromise = new Promise((resolve) => {
+  document.addEventListener('click', () => {
```

Comment:
Nested event listeners inside the thirdPromise. This could lead to unexpected behavior and memory leaks if not handled correctly. Consider redesigning the logic to avoid unnecessary nesting.